From 3a065628c897902e5568638c01f8979b742d9912 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 12 Jun 2025 16:12:37 +0200
Subject: [PATCH 187/217] genirq: irq_pipeline: prevent forced threading of oob
 IRQ

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/irq/manage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/irq/manage.c b/kernel/irq/manage.c
index 3a65fa887..716057c01 100644
--- a/kernel/irq/manage.c
+++ b/kernel/irq/manage.c
@@ -1426,7 +1426,7 @@ static int irq_setup_forced_threading(struct irqaction *new)
 {
 	if (!force_irqthreads())
 		return 0;
-	if (new->flags & (IRQF_NO_THREAD | IRQF_PERCPU | IRQF_ONESHOT))
+	if (new->flags & (IRQF_NO_THREAD | IRQF_PERCPU | IRQF_ONESHOT | IRQF_OOB))
 		return 0;
 
 	/*
-- 
2.30.2

