From 7056048eb1f21574e1ed510218aeedbebba08cd5 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 28 Sep 2024 18:36:24 +0200
Subject: [PATCH 0165/1010] net: dovetail: select PAGE_POOL from PAGE_POOL_OOB

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 net/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/Kconfig b/net/Kconfig
index a0b464dfa4..ac6193ea54 100644
--- a/net/Kconfig
+++ b/net/Kconfig
@@ -439,6 +439,7 @@ config PAGE_POOL
 
 config PAGE_POOL_OOB
 	def_bool n
+	select PAGE_POOL
 
 config PAGE_POOL_STATS
 	default n
-- 
2.30.2

