From 4eced08d1d984016398432c7a7ef5a24b886b8f9 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 16 Oct 2024 15:48:29 +0200
Subject: [PATCH 0167/1010] net: dovetail: fix inline documentation

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/linux/netdevice.h | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/include/linux/netdevice.h b/include/linux/netdevice.h
index f9b4144385..559edbc4e7 100644
--- a/include/linux/netdevice.h
+++ b/include/linux/netdevice.h
@@ -1410,9 +1410,6 @@ struct netdev_net_notifier {
  * void	(*ndo_disable_oob)(struct net_device *dev);
  *	Turn off out-of-band I/O handling, reverting the effect of ndo_enable_oob().
  *	This handler is optional.
- * void	(*ndo_disable_oob)(struct net_device *dev);
- *	Turn off out-of-band I/O handling, reverting the effect of ndo_enable_oob().
- *	This handler is optional.
  */
 struct net_device_ops {
 	int			(*ndo_init)(struct net_device *dev);
@@ -4314,8 +4311,8 @@ bool netif_receive_oob(struct sk_buff *skb);
  *	netif_oob_diversion - is the ingress traffic diverted for out-of-band handling?
  *	@dev: network device
  *
- * Check if the RX packets are currently diverted to a companion
- * kernel for out-of-band handling.
+ * Check if the RX packets to this (real) device are currently
+ * diverted to a companion kernel for out-of-band handling.
  */
 static inline bool netif_oob_diversion(const struct net_device *dev)
 {
-- 
2.30.2

