From fa5d2aaace0e180411beb3044675905dfd233a03 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Sun, 27 Apr 2025 15:43:24 +0200
Subject: [PATCH 169/985] net: dovetail: add stage exclusion to tx_lock

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 include/linux/netdevice.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/linux/netdevice.h b/include/linux/netdevice.h
index 772890cfc..7dec86b43 100644
--- a/include/linux/netdevice.h
+++ b/include/linux/netdevice.h
@@ -4576,6 +4576,7 @@ static inline void __netif_tx_lock(struct netdev_queue *txq, int cpu)
 	spin_lock(&txq->_xmit_lock);
 	/* Pairs with READ_ONCE() in __dev_queue_xmit() */
 	WRITE_ONCE(txq->xmit_lock_owner, cpu);
+	netif_tx_lock_oob(txq);
 }
 
 static inline bool __netif_tx_acquire(struct netdev_queue *txq)
@@ -4594,6 +4595,7 @@ static inline void __netif_tx_lock_bh(struct netdev_queue *txq)
 	spin_lock_bh(&txq->_xmit_lock);
 	/* Pairs with READ_ONCE() in __dev_queue_xmit() */
 	WRITE_ONCE(txq->xmit_lock_owner, smp_processor_id());
+	netif_tx_lock_oob(txq);
 }
 
 static inline bool __netif_tx_trylock(struct netdev_queue *txq)
@@ -4603,12 +4605,14 @@ static inline bool __netif_tx_trylock(struct netdev_queue *txq)
 	if (likely(ok)) {
 		/* Pairs with READ_ONCE() in __dev_queue_xmit() */
 		WRITE_ONCE(txq->xmit_lock_owner, smp_processor_id());
+		netif_tx_lock_oob(txq);
 	}
 	return ok;
 }
 
 static inline void __netif_tx_unlock(struct netdev_queue *txq)
 {
+	netif_tx_unlock_oob(txq);
 	/* Pairs with READ_ONCE() in __dev_queue_xmit() */
 	WRITE_ONCE(txq->xmit_lock_owner, -1);
 	spin_unlock(&txq->_xmit_lock);
@@ -4616,6 +4620,7 @@ static inline void __netif_tx_unlock(struct netdev_queue *txq)
 
 static inline void __netif_tx_unlock_bh(struct netdev_queue *txq)
 {
+	netif_tx_unlock_oob(txq);
 	/* Pairs with READ_ONCE() in __dev_queue_xmit() */
 	WRITE_ONCE(txq->xmit_lock_owner, -1);
 	spin_unlock_bh(&txq->_xmit_lock);
-- 
2.30.2

