From 035ddf5ec5c264a81fc8918fc90185db4f6b6408 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 12 Jun 2025 13:05:22 +0200
Subject: [PATCH 0185/1013] genirq: irq_pipeline: fix report format on
 unhandled irq

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/irq/pipeline.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/irq/pipeline.c b/kernel/irq/pipeline.c
index 64fcf4b399c0..cc352bb07494 100644
--- a/kernel/irq/pipeline.c
+++ b/kernel/irq/pipeline.c
@@ -879,10 +879,10 @@ static void handle_unexpected_irq(struct irq_desc *desc, irqreturn_t ret)
 		     !raw_spin_is_locked(&desc->lock));
 
 	if (ret != IRQ_NONE) {
-		printk(KERN_ERR "out-of-band irq event %d: bogus return value %x\n",
+		printk(KERN_ERR "out-of-band IRQ%d: bogus return value %#x\n",
 		       irq, ret);
 		for_each_action_of_desc(desc, action)
-			printk(KERN_ERR "[<%p>] %pf",
+			printk(KERN_ERR "[<%px>] %pS",
 			       action->handler, action->handler);
 		printk(KERN_CONT "\n");
 		return;
-- 
2.30.2

