From 1c88deb604902fd6a73475db593dd0eb397db054 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 25 Jun 2025 09:01:59 +0200
Subject: [PATCH 208/985] net: dovetail: init oob markers at skb setup

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/linux/skbuff.h | 19 +++++++++++++++++++
 net/core/skbuff.c      |  5 +++++
 2 files changed, 24 insertions(+)

diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index 0808b27b8..2790f237e 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -5270,6 +5270,16 @@ struct sk_buff *get_oob_skb(void);
 
 void put_oob_skb(struct sk_buff *skb);
 
+/**
+ * skb_init_inband - Set up the oob markers appropriately for an
+ * in-band skb.
+ */
+static inline void skb_init_inband(struct sk_buff *skb)
+{
+	skb->oob = 0;
+	skb->oob_released = 0;
+}
+
 /**
  * skb_is_oob - Whether the buffer shell was allocated from the oob
  * pool. Caution: this is distinct from a skb which storage is managed
@@ -5295,6 +5305,11 @@ static inline void skb_mark_oob_released(struct sk_buff *skb)
 	skb->oob_released = 1;
 }
 
+static inline void skb_clear_oob_released(struct sk_buff *skb)
+{
+	skb->oob_released = 0;
+}
+
 /**
  * skb_is_oob_timestamped - Whether the buffer is timestamped by a
  * companion core, in which case the timestamps should be stored in
@@ -5359,6 +5374,10 @@ static inline bool __skb_oob_free_head(struct sk_buff *skb)
 
 #else  /* !CONFIG_NET_OOB */
 
+static inline void skb_init_inband(struct sk_buff *skb)
+{
+}
+
 static inline bool skb_is_oob(const struct sk_buff *skb)
 {
 	return false;
diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index 5183aa9f0..74dda4e49 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -655,6 +655,7 @@ struct sk_buff *get_oob_skb(void)
 		raw_spin_unlock_irqrestore(&c->lock, flags);
 		memset(skb, 0, offsetof(struct sk_buff, tail));
 		skb_mark_oob(skb);
+		skb_clear_oob_released(skb);
 	} else {
 		raw_spin_unlock_irqrestore(&c->lock, flags);
 	}
@@ -1773,6 +1774,7 @@ static struct sk_buff *__skb_clone(struct sk_buff *n, struct sk_buff *skb)
 	C(data);
 	C(truesize);
 	refcount_set(&n->users, 1);
+	skb_init_inband(n);
 
 	atomic_inc(&(skb_shinfo(skb)->dataref));
 	skb->cloned = 1;
@@ -2498,6 +2500,7 @@ int pskb_expand_head(struct sk_buff *skb, int nhead, int ntail,
 	skb->hdr_len  = 0;
 	skb->nohdr    = 0;
 	atomic_set(&skb_shinfo(skb)->dataref, 1);
+	skb_init_inband(skb);
 
 	skb_metadata_clear(skb);
 
@@ -6879,6 +6882,7 @@ static int pskb_carve_inside_header(struct sk_buff *skb, const u32 off,
 	skb->hdr_len = 0;
 	skb->nohdr = 0;
 	atomic_set(&skb_shinfo(skb)->dataref, 1);
+	skb_init_inband(skb);
 
 	return 0;
 }
@@ -7015,6 +7019,7 @@ static int pskb_carve_inside_nonlinear(struct sk_buff *skb, const u32 off,
 	skb->len -= off;
 	skb->data_len = skb->len;
 	atomic_set(&skb_shinfo(skb)->dataref, 1);
+	skb_init_inband(skb);
 
 	return 0;
 }
-- 
2.30.2

