From f5650dec0feb63a0a3a4170992b78ce62825012b Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Tue, 29 Apr 2025 15:25:04 +0200
Subject: [PATCH 0210/1010] net: dovetail: increase size of oob skb pool

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 net/core/skbuff.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/core/skbuff.c b/net/core/skbuff.c
index c080b5ccc0..100e46116f 100644
--- a/net/core/skbuff.c
+++ b/net/core/skbuff.c
@@ -301,7 +301,7 @@ static void __build_skb_around(struct sk_buff *skb, void *data,
 
 #ifdef CONFIG_NET_OOB
 
-unsigned int sysctl_max_oob_skb __read_mostly = 1024;
+unsigned int sysctl_max_oob_skb __read_mostly = 2048;
 EXPORT_SYMBOL(sysctl_max_oob_skb);
 
 __weak void free_skb_oob(struct sk_buff *skb)
-- 
2.30.2

