From d9caa7dcb29f61bb5c1b89c31d760d0e303ec0e4 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 18 Sep 2025 11:03:15 +0200
Subject: [PATCH 216/985] net/page_pool: dovetail: fix build (missing call
 argument)

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 net/core/page_pool.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/core/page_pool.c b/net/core/page_pool.c
index df8e28940..d7de177af 100644
--- a/net/core/page_pool.c
+++ b/net/core/page_pool.c
@@ -601,7 +601,7 @@ static int page_pool_init_cache(struct page_pool *pool)
 		for (i = 0; i < nr_pages; i++) {
 			netmem = pool->alloc.cache[i];
 			if ((pp_flags & PP_FLAG_DMA_MAP) &&
-				unlikely(!page_pool_dma_map(pool, netmem))) {
+			    unlikely(!page_pool_dma_map(pool, netmem, GFP_KERNEL))) {
 				nr_pages = i;
 				goto fail;
 			}
-- 
2.30.2

