From dbfd78e72a2b0fcad978bf8122152e94257b9fb3 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 27 Jan 2019 10:52:28 +0100
Subject: [PATCH 0227/1013] evl/poller: fix initializers

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evenless/poller.h | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/include/evenless/poller.h b/include/evenless/poller.h
index 5ca68e9200c3..b78910daee02 100644
--- a/include/evenless/poller.h
+++ b/include/evenless/poller.h
@@ -16,11 +16,9 @@
 #include <evenless/factory.h>
 #include <uapi/evenless/poller.h>
 
-struct oob_poll_wait;
-
 #define EVL_POLLHEAD_INITIALIZER(__name) {				\
 		.watchpoints = LIST_HEAD_INIT((__name).watchpoints),	\
-		__HARD_SPIN_LOCK_INIT((__name).__lock),			\
+		lock = __HARD_SPIN_LOCK_INITIALIZER(__name),		\
 	}
 
 struct evl_poll_head {
@@ -32,7 +30,7 @@ static inline
 void evl_init_poll_head(struct evl_poll_head *head)
 {
 	INIT_LIST_HEAD(&head->watchpoints);
-	raw_spin_lock_init(&head->lock);
+	hard_spin_lock_init(&head->lock);
 }
 
 void evl_poll_watch(struct evl_poll_head *head,
-- 
2.30.2

