From 74b09355968d8cd16caf97cb054ecd21f5c94fe2 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 3 Feb 2019 11:00:42 +0100
Subject: [PATCH 0250/1013] evl/wait: call evl_schedule() lock-free

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evenless/wait.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/kernel/evenless/wait.c b/kernel/evenless/wait.c
index af3b049398e5..56e71dfe7341 100644
--- a/kernel/evenless/wait.c
+++ b/kernel/evenless/wait.c
@@ -40,7 +40,6 @@ int evl_wait_timeout(struct evl_wait_queue *wq, ktime_t timeout,
 {
 	struct evl_thread *curr = evl_current();
 	unsigned long flags;
-	int ret;
 
 	if (IS_ENABLED(CONFIG_EVENLESS_DEBUG_MUTEX_SLEEP) &&
 		atomic_read(&curr->inband_disable_count) &&
@@ -57,12 +56,12 @@ int evl_wait_timeout(struct evl_wait_queue *wq, ktime_t timeout,
 		list_add_priff(curr, &wq->wait_list, wprio, wait_next);
 
 	evl_sleep_on(timeout, timeout_mode, wq->clock, &wq->wchan);
-	evl_schedule();
-	ret = curr->info & (T_RMID|T_TIMEO|T_BREAK);
 
 	xnlock_put_irqrestore(&nklock, flags);
 
-	return ret;
+	evl_schedule();
+
+	return curr->info & (T_RMID|T_TIMEO|T_BREAK);
 }
 EXPORT_SYMBOL_GPL(evl_wait_timeout);
 
-- 
2.30.2

