From 161fa7fefdd65ee3a3be2fd28edf1f8eb1353a33 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 3 Feb 2019 11:01:22 +0100
Subject: [PATCH 0251/1013] evl/xbuf: call evl_schedule() lock-free

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evenless/xbuf.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/kernel/evenless/xbuf.c b/kernel/evenless/xbuf.c
index cc3be3f12707..ae0cf9ee9ca3 100644
--- a/kernel/evenless/xbuf.c
+++ b/kernel/evenless/xbuf.c
@@ -190,7 +190,6 @@ static ssize_t do_xbuf_read(struct xbuf_ring *ring,
 		 * we freed enough room to post its message.
 		 */
 		ring->unblock_output(ring);
-		evl_schedule();
 		goto out;
 	wait:
 		if (len > ring->bufsz)
@@ -218,6 +217,8 @@ static ssize_t do_xbuf_read(struct xbuf_ring *ring,
 out:
 	xnlock_put_irqrestore(&nklock, flags);
 
+	evl_schedule();
+
 	return ret;
 }
 
@@ -295,7 +296,6 @@ static ssize_t do_xbuf_write(struct xbuf_ring *ring,
 			ring->clear_pollable(ring, POLLOUT|POLLWRNORM);
 
 		ring->signal_input(ring);
-		evl_schedule();
 		goto out;
 	wait:
 		if (f_flags & O_NONBLOCK) {
@@ -314,6 +314,8 @@ static ssize_t do_xbuf_write(struct xbuf_ring *ring,
 out:
 	xnlock_put_irqrestore(&nklock, flags);
 
+	evl_schedule();
+
 	return ret;
 }
 
-- 
2.30.2

