From 8dac0e3eb487d03af547527e8e75a3caabbcd377 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 9 Feb 2019 15:37:04 +0100
Subject: [PATCH 0282/1013] evl/monitor: signal: send ESRCH if target is not
 found

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evenless/monitor.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/evenless/monitor.c b/kernel/evenless/monitor.c
index 14f586704963..40abfebf6328 100644
--- a/kernel/evenless/monitor.c
+++ b/kernel/evenless/monitor.c
@@ -58,7 +58,7 @@ int evl_signal_monitor_targeted(struct evl_thread *target, int monfd)
 	struct evl_monitor *event;
 	struct evl_file *efilp;
 	unsigned long flags;
-	int ret = -EAGAIN;
+	int ret = -ESRCH;
 
 	event = get_monitor_by_fd(monfd, &efilp);
 	if (event == NULL)
-- 
2.30.2

