From cd2b781d6d7a85f3308b7cc12efb93bedc416ad7 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 15 Feb 2019 16:32:17 +0100
Subject: [PATCH 0303/1013] evl/monitor: unblock waiters on release

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evenless/monitor.c | 27 +++++++++++++++++++++------
 1 file changed, 21 insertions(+), 6 deletions(-)

diff --git a/kernel/evenless/monitor.c b/kernel/evenless/monitor.c
index caaf2b77c701..fde244222206 100644
--- a/kernel/evenless/monitor.c
+++ b/kernel/evenless/monitor.c
@@ -181,11 +181,14 @@ static int __enter_monitor(struct evl_monitor *gate,
 
 	tmode = timeout ? EVL_ABS : EVL_REL;
 	info = evl_lock_mutex_timeout(&gate->lock, timeout, tmode);
-	if (info) {
-		if (info & T_BREAK)
-			return -EINTR;
-		return info & T_TIMEO ? -ETIMEDOUT : -EINVAL;
-	}
+	if (info & T_BREAK)
+		return -EINTR;
+
+	if (info & T_RMID)
+		return -EIDRM;
+
+	if (info & T_TIMEO)
+		return -ETIMEDOUT;
 
 	return 0;
 }
@@ -472,9 +475,21 @@ static long monitor_oob_ioctl(struct file *filp, unsigned int cmd,
 	return ret;
 }
 
+static int monitor_release(struct inode *inode, struct file *filp)
+{
+	struct evl_monitor *mon = element_of(filp, struct evl_monitor);
+
+	if (mon->type == EVL_MONITOR_EV)
+		evl_flush_wait(&mon->wait_queue, T_RMID);
+	else
+		evl_flush_mutex(&mon->lock, T_RMID);
+
+	return evl_release_element(inode, filp);
+}
+
 static const struct file_operations monitor_fops = {
 	.open		= evl_open_element,
-	.release	= evl_release_element,
+	.release	= monitor_release,
 	.unlocked_ioctl	= monitor_ioctl,
 	.oob_ioctl	= monitor_oob_ioctl,
 };
-- 
2.30.2

