From 2b56d5526c1f7ce86427553bcb4e3aef12e81bff Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 15 Feb 2019 16:40:00 +0100
Subject: [PATCH 0305/1013] evl/xbuf: unblock waiters on release

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evenless/xbuf.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/kernel/evenless/xbuf.c b/kernel/evenless/xbuf.c
index 1b9faaadf79d..9e283ec22a18 100644
--- a/kernel/evenless/xbuf.c
+++ b/kernel/evenless/xbuf.c
@@ -571,9 +571,19 @@ static __poll_t xbuf_oob_poll(struct file *filp,
 	return ready;
 }
 
+static int xbuf_release(struct inode *inode, struct file *filp)
+{
+	struct evl_xbuf *xbuf = element_of(filp, struct evl_xbuf);
+
+	evl_flush_flag(&xbuf->obnd.i_event, T_RMID);
+	evl_flush_flag(&xbuf->ibnd.o_event, T_RMID);
+
+	return evl_release_element(inode, filp);
+}
+
 static const struct file_operations xbuf_fops = {
 	.open		= evl_open_element,
-	.release	= evl_release_element,
+	.release	= xbuf_release,
 	.unlocked_ioctl	= xbuf_ioctl,
 	.read		= xbuf_read,
 	.write		= xbuf_write,
-- 
2.30.2

