From b983d04fbee5a56c9c095f4e0db6172ad9b613c2 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 9 Feb 2019 15:33:46 +0100
Subject: [PATCH 0315/1010] evl/factory: send ENXIO if core is unavailable

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evenless/factory.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/evenless/factory.c b/kernel/evenless/factory.c
index d7822d39f3..f8ae829be1 100644
--- a/kernel/evenless/factory.c
+++ b/kernel/evenless/factory.c
@@ -300,7 +300,7 @@ static long ioctl_clone_device(struct file *filp, unsigned int cmd,
 		return -ENOTTY;
 
 	if (!evl_is_running())
-		return -EAGAIN;
+		return -ENXIO;
 
 	if (e)
 		return -EBUSY;
-- 
2.30.2

