From 4cf9e3b491434bb095036c32e1a33447cd571fbf Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 16 Feb 2019 18:20:05 +0100
Subject: [PATCH 0348/1010] evl/syscall: shorten syscall name prefix

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evenless/syscall.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/evenless/syscall.c b/kernel/evenless/syscall.c
index 865eeb381b..cd5fab3ea1 100644
--- a/kernel/evenless/syscall.c
+++ b/kernel/evenless/syscall.c
@@ -31,7 +31,7 @@
 #endif
 
 #define EVENLESS_SYSCALL(__name, __args)	\
-	long EvEnLeSs_ ## __name __args
+	long EVL_ ## __name __args
 
 #define SYSCALL_PROPAGATE   0
 #define SYSCALL_STOP        1
@@ -317,12 +317,12 @@ static EVENLESS_SYSCALL(ioctl, (int fd, unsigned int request,
 	return ret;
 }
 
-static int EvEnLeSs_ni(void)
+static int EVL_ni(void)
 {
 	return -ENOSYS;
 }
 
-#define __syshand__(__name)	((evl_syshand)(EvEnLeSs_ ## __name))
+#define __syshand__(__name)	((evl_syshand)(EVL_ ## __name))
 
 #define __EVENLESS_CALL_ENTRIES			\
 	__EVENLESS_CALL_ENTRY(read)		\
-- 
2.30.2

