From df229877b5715039ad34a756cace6bc10c084bfd Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 18 Mar 2019 16:11:30 +0100
Subject: [PATCH 0350/1013] evl/assert: fix SMP build

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/assert.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/evl/assert.h b/include/evl/assert.h
index 1eecf6ca8985..261942a59b5b 100644
--- a/include/evl/assert.h
+++ b/include/evl/assert.h
@@ -28,8 +28,7 @@
 #define EVL_WARN_ON_SMP(__subsys, __cond)		\
 	EVL_WARN_ON(__subsys, __cond)
 #else
-#define EVL_WARN_ON_SMP(__subsys, __cond)		\
-	do { } while (0)
+#define EVL_WARN_ON_SMP(__subsys, __cond)  0
 #endif
 
 #define oob_context_only()	EVL_WARN_ON_ONCE(CORE, running_inband())
-- 
2.30.2

