From 5c8be6e14a321441d8d0955cd9e5cafff442f8c6 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 9 May 2019 16:29:10 +0200
Subject: [PATCH 369/985] evl/trace: fix evl_wakeup_thread event trace

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/trace/events/evl.h | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/include/trace/events/evl.h b/include/trace/events/evl.h
index deaca0a65..8c781ab4e 100644
--- a/include/trace/events/evl.h
+++ b/include/trace/events/evl.h
@@ -337,25 +337,26 @@ TRACE_EVENT(evl_sleep_on,
 );
 
 TRACE_EVENT(evl_wakeup_thread,
-	TP_PROTO(struct evl_thread *thread, int mask, int reason),
-	TP_ARGS(thread, mask, reason),
+	TP_PROTO(struct evl_thread *thread, int mask, int info),
+	TP_ARGS(thread, mask, info),
 
 	TP_STRUCT__entry(
 		__string(name, thread->name)
 		__field(pid_t, pid)
 		__field(int, mask)
-		__field(int, reason)
+		__field(int, info)
 	),
 
 	TP_fast_assign(
 		__assign_str(name, thread->name);
 		__entry->pid = evl_get_inband_pid(thread);
 		__entry->mask = mask;
+		__entry->info = info;
 	),
 
-	TP_printk("name=%s pid=%d mask=%#x reason=%#x",
+	TP_printk("name=%s pid=%d mask=%#x info=%#x",
 		__get_str(name), __entry->pid,
-		__entry->mask, __entry->reason)
+		__entry->mask, __entry->info)
 );
 
 TRACE_EVENT(evl_hold_thread,
-- 
2.30.2

