From ed53d74675be5b02c05afe7155cb8fcdeddea7f0 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 21 Mar 2019 12:57:37 +0100
Subject: [PATCH 0388/1010] evl/stat: use xchg() for swapping pointers

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/stat.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/evl/stat.h b/include/evl/stat.h
index f669f2ba9f..3a5d340d05 100644
--- a/include/evl/stat.h
+++ b/include/evl/stat.h
@@ -69,8 +69,7 @@ static inline void evl_reset_account(struct evl_account *account)
 	({								\
 		struct evl_account *__prev;				\
 		__prev = (struct evl_account *)				\
-			atomic_long_xchg(&(__rq)->current_account,	\
-					(long)(__new_account));		\
+			xchg(&(__rq)->current_account, (__new_account)); \
 		__prev;							\
 	})
 
-- 
2.30.2

