From f11a6bba86be7e1b56d67925c91af60455c92058 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 31 May 2019 15:11:51 +0200
Subject: [PATCH 0391/1013] evl/tick: clearify, update comment

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/tick.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/kernel/evl/tick.c b/kernel/evl/tick.c
index a80fc924fe7f..f9780664cb5c 100644
--- a/kernel/evl/tick.c
+++ b/kernel/evl/tick.c
@@ -105,11 +105,11 @@ static void clock_event_handler(struct clock_event_device *dummy)
 	evl_announce_tick(&evl_mono_clock);
 
 	/*
-	 * If a real-time thread was preempted by this clock
-	 * interrupt, any transition to the root thread will cause a
-	 * in-band tick to be propagated by evl_schedule() from
-	 * irq_finish_head(), so we only need to propagate the in-band
-	 * tick in case the root thread was preempted.
+	 * If an EVL thread was preempted by this clock event, any
+	 * transition to the root thread will cause a pending in-band
+	 * tick to be propagated by evl_schedule() from
+	 * exit_oob_irq(), so we may have to propagate the in-band
+	 * tick immediately only if the root thread was preempted.
 	 */
 	if ((this_rq->lflags & RQ_TPROXY) && (this_rq->curr->state & T_ROOT))
 		evl_notify_proxy_tick(this_rq);
-- 
2.30.2

