From b6675b9f97ae33f73f6334c400842e1542edc677 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 22 Apr 2019 18:16:32 +0200
Subject: [PATCH 0392/1010] uapi: evl: rename sigshadow to sigevl

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/uapi/evl/signal.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/uapi/evl/signal.h b/include/uapi/evl/signal.h
index 74e53bc106..e98759c805 100644
--- a/include/uapi/evl/signal.h
+++ b/include/uapi/evl/signal.h
@@ -10,9 +10,9 @@
 #define _EVL_UAPI_SIGNAL_H
 
 #define SIGEVL				SIGWINCH
-#define sigshadow_action(code)		((code) & 0xff)
-#define sigshadow_arg(code)		(((code) >> 8) & 0xff)
-#define sigshadow_int(action, arg)	((action) | ((arg) << 8))
+#define sigevl_action(code)		((code) & 0xff)
+#define sigevl_arg(code)		(((code) >> 8) & 0xff)
+#define sigevl_int(action, arg)		((action) | ((arg) << 8))
 
 /* SIGEVL action codes. */
 #define SIGEVL_ACTION_HOME		1
-- 
2.30.2

