From 916c2ec23604547258e88060aa7c922d8aef54a1 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 4 May 2019 11:42:04 +0200
Subject: [PATCH 0395/1010] evl/timer: fix comment

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/timer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/evl/timer.c b/kernel/evl/timer.c
index 70cf1b8275..6fba4243a5 100644
--- a/kernel/evl/timer.c
+++ b/kernel/evl/timer.c
@@ -299,7 +299,7 @@ void __evl_set_timer_rq(struct evl_timer *timer,
 
 	/*
 	 * Figure out which CPU is best suited for managing this
-	 * timer, preferably picking xnsched_cpu(sched) if the ticking
+	 * timer, preferably picking evl_rq_cpu(rq) if the ticking
 	 * device moving the timer clock beats on that CPU. Otherwise,
 	 * pick the first CPU from the clock affinity mask if set. If
 	 * not, the timer is backed by a global device with no
-- 
2.30.2

