From 441ad9c27249cf89e89659508dfe1cb84219e560 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 7 Jun 2019 15:51:11 +0200
Subject: [PATCH 397/985] evl/trace: rename worst case latency event

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/trace/events/evl.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/trace/events/evl.h b/include/trace/events/evl.h
index bd42f8f9f..33449630a 100644
--- a/include/trace/events/evl.h
+++ b/include/trace/events/evl.h
@@ -858,7 +858,7 @@ TRACE_EVENT(evl_latspot,
 	TP_fast_assign(
 		__entry->latmax_ns = latmax_ns;
 	),
-	TP_printk("** latency spot: %d.%.3d us **",
+	TP_printk("** latency peak: %d.%.3d us **",
 		  __entry->latmax_ns / 1000,
 		  __entry->latmax_ns % 1000)
 );
-- 
2.30.2

