From f240f5a7b57b5a21d0cc4aae501b013312e748e0 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 30 May 2019 15:51:22 +0200
Subject: [PATCH 0420/1010] uapi/evl: fix thread state labels

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/uapi/evl/thread.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/include/uapi/evl/thread.h b/include/uapi/evl/thread.h
index 574394d4e9..b7acf7b641 100644
--- a/include/uapi/evl/thread.h
+++ b/include/uapi/evl/thread.h
@@ -61,8 +61,10 @@
  * 'S' -> Forcibly suspended
  * 'w'/'W' -> Blocked with/without timeout
  * 'D' -> Delayed
- * 'R' -> Runnable
+ * 'p' -> Periodic timeline
+ * 'R' -> Ready to run
  * 'U' -> Dormant
+ * 'Z' -> Zombie
  * 'X' -> Running in-band
  * 'H' -> Held in emergency
  * 'b' -> Priority boost undergoing
@@ -70,7 +72,7 @@
  * 'r' -> Undergoes round-robin
  * 't' -> SIGDEBUG notifications enabled
  */
-#define EVL_THREAD_STATE_LABELS  "SWDRU.XHbTrt...."
+#define EVL_THREAD_STATE_LABELS  "SWDpRUZXHbTrt...."
 
 struct evl_user_window {
 	__u32 state;
-- 
2.30.2

