From bd82149b506605e1098dd378b4e427c00b1d85e4 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Fri, 28 Jun 2019 11:08:41 +0200
Subject: [PATCH 425/985] evl/control: export device name as symbol

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/uapi/evl/control.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/uapi/evl/control.h b/include/uapi/evl/control.h
index 424d2119e..9bc017e90 100644
--- a/include/uapi/evl/control.h
+++ b/include/uapi/evl/control.h
@@ -12,6 +12,8 @@
 
 #define EVL_ABI_LEVEL  5
 
+#define EVL_CONTROL_DEV  "/dev/evl/control"
+
 struct evl_core_info {
 	__u32 abi_level;
 	__u32 fpu_features;
-- 
2.30.2

