From 0c30458afc38ef344252f7b28cbe08167ef09d4a Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 1 Jul 2019 11:01:16 +0200
Subject: [PATCH 431/985] evl/timerfd: enable get/set from in-band context

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/clock.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/kernel/evl/clock.c b/kernel/evl/clock.c
index c8ab6a688..6d97fed08 100644
--- a/kernel/evl/clock.c
+++ b/kernel/evl/clock.c
@@ -606,8 +606,8 @@ static bool read_timerfd_event(struct evl_timerfd *timerfd)
 	return false;
 }
 
-static long timerfd_oob_ioctl(struct file *filp,
-			unsigned int cmd, unsigned long arg)
+static long timerfd_common_ioctl(struct file *filp,
+				unsigned int cmd, unsigned long arg)
 {
 	struct evl_timerfd *timerfd = filp->private_data;
 	struct evl_timerfd_setreq sreq, __user *u_sreq;
@@ -702,9 +702,10 @@ static int timerfd_release(struct inode *inode, struct file *filp)
 
 static const struct file_operations timerfd_fops = {
 	.release	= timerfd_release,
-	.oob_ioctl	= timerfd_oob_ioctl,
+	.oob_ioctl	= timerfd_common_ioctl,
 	.oob_read	= timerfd_oob_read,
 	.oob_poll	= timerfd_oob_poll,
+	.unlocked_ioctl	= timerfd_common_ioctl,
 };
 
 static int new_timerfd(struct evl_clock *clock)
-- 
2.30.2

