From 92c064add5e1357577c1886bcde0a5fb388e8b2a Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 13 Jul 2019 21:39:15 +0200
Subject: [PATCH 447/985] evl/thread: drop obsolete comment

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/thread.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/kernel/evl/thread.c b/kernel/evl/thread.c
index 956d7bda8..10b98d13c 100644
--- a/kernel/evl/thread.c
+++ b/kernel/evl/thread.c
@@ -1736,11 +1736,6 @@ void resume_oob_task(struct task_struct *p) /* hw IRQs off */
 {
 	struct evl_thread *thread = evl_thread_from_task(p);
 
-	/*
-	 * We fire the handler before the thread is migrated, so that
-	 * thread->rq does not change between paired invocations of
-	 * relax_thread/harden handlers.
-	 */
 	xnlock_get(&nklock);
 	if (affinity_ok(p))
 		evl_release_thread(thread, T_INBAND, 0);
-- 
2.30.2

