From 455f826c0523226d482bc52833294825cc99b8eb Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 21 Jul 2019 22:41:23 +0200
Subject: [PATCH 0451/1013] drivers/evl: latmus: care for signedness in overrun
 calculation

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 drivers/evl/latmus.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/evl/latmus.c b/drivers/evl/latmus.c
index 97c630108327..286fc3b830ac 100644
--- a/drivers/evl/latmus.c
+++ b/drivers/evl/latmus.c
@@ -185,7 +185,8 @@ static int add_measurement_sample(struct latmus_runner *runner,
 	state->sum += delta;
 	state->ideal = ktime_add(state->ideal, period);
 
-	while (delta > ktime_to_ns(period)) { /* period > 0 */
+	while (delta > 0 &&
+		(unsigned int)delta > ktime_to_ns(period)) { /* period > 0 */
 		state->overruns++;
 		state->ideal = ktime_add(state->ideal, period);
 		delta -= ktime_to_ns(period);
-- 
2.30.2

