From 0222126892e084539208cf89856470802452dc96 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 3 Aug 2019 16:50:25 +0200
Subject: [PATCH 463/985] evl/thread: replace static cpumask w/ dynamic var

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/thread.c | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/kernel/evl/thread.c b/kernel/evl/thread.c
index 3feb011f2..dacde886c 100644
--- a/kernel/evl/thread.c
+++ b/kernel/evl/thread.c
@@ -133,9 +133,11 @@ int evl_init_thread(struct evl_thread *thread,
 		const char *fmt, ...)
 {
 	int flags = iattr->flags & ~T_SUSP, ret, gravity;
-	struct cpumask affinity;
+	cpumask_var_t affinity;
 	va_list args;
 
+	inband_context_only();
+
 	if (!(flags & T_ROOT))
 		flags |= T_DORMANT | T_INBAND;
 
@@ -148,11 +150,15 @@ int evl_init_thread(struct evl_thread *thread,
 	 * of the supported CPUs. This CPU may change in
 	 * pin_to_initial_cpu().
 	 */
-	if (rq == NULL) {
-		cpumask_and(&affinity, &iattr->affinity, &evl_cpu_affinity);
-		if (cpumask_empty(&affinity))
+	if (!rq) {
+		if (!alloc_cpumask_var(&affinity, GFP_KERNEL))
+			return -ENOMEM;
+		cpumask_and(affinity, &iattr->affinity, &evl_cpu_affinity);
+		if (!cpumask_empty(affinity))
+			rq = evl_cpu_rq(cpumask_first(affinity));
+		free_cpumask_var(affinity);
+		if (!rq)
 			return -EINVAL;
-		rq = evl_cpu_rq(cpumask_first(&affinity));
 	}
 
 	va_start(args, fmt);
-- 
2.30.2

