From cd4a98c5c96dc574b520accba4b82cef6ee02d2f Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 14 Jul 2019 22:32:19 +0200
Subject: [PATCH 0483/1010] evl/clock: drop unused variable in UP build

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/clock.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/kernel/evl/clock.c b/kernel/evl/clock.c
index c8428b0969..81ad32141c 100644
--- a/kernel/evl/clock.c
+++ b/kernel/evl/clock.c
@@ -358,7 +358,6 @@ void evl_core_tick(struct clock_event_device *dummy) /* hard irqs off */
 
 void evl_announce_tick(struct evl_clock *clock) /* hard irqs off */
 {
-	struct evl_rq *this_rq = this_evl_rq();
 	struct evl_timerbase *tmb;
 
 #ifdef CONFIG_SMP
@@ -367,7 +366,7 @@ void evl_announce_tick(struct evl_clock *clock) /* hard irqs off */
 	 * particular IRQ affinity, in which case the associated
 	 * timers will be queued to CPU0.
 	 */
-	if (!cpumask_test_cpu(evl_rq_cpu(this_rq), &clock->affinity))
+	if (!cpumask_test_cpu(evl_rq_cpu(this_evl_rq()), &clock->affinity))
 		tmb = evl_percpu_timers(clock, 0);
 	else
 #endif
-- 
2.30.2

