From b53195ebed2bb144179a6e248f856714ab4293f7 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 8 Oct 2019 16:13:35 +0200
Subject: [PATCH 490/985] evl/monitor: commit PP unlocked in tryenter()

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/monitor.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/kernel/evl/monitor.c b/kernel/evl/monitor.c
index 8b2fb5baa..5dcd1e496 100644
--- a/kernel/evl/monitor.c
+++ b/kernel/evl/monitor.c
@@ -92,7 +92,7 @@ int evl_signal_monitor_targeted(struct evl_thread *target, int monfd)
 	return ret;
 }
 
-void __evl_commit_monitor_ceiling(void)  /* nklock held, irqs off, OOB */
+void __evl_commit_monitor_ceiling(void)
 {
 	struct evl_thread *curr = evl_current();
 	struct evl_monitor *gate;
@@ -233,11 +233,13 @@ static int tryenter_monitor(struct evl_monitor *gate)
 	unsigned long flags;
 	int ret;
 
+	no_ugly_lock();
+
 	if (gate->type != EVL_MONITOR_GATE)
 		return -EINVAL;
 
-	xnlock_get_irqsave(&nklock, flags);
 	evl_commit_monitor_ceiling();
+	xnlock_get_irqsave(&nklock, flags);
 	ret = evl_trylock_mutex(&gate->lock);
 	xnlock_put_irqrestore(&nklock, flags);
 
-- 
2.30.2

