From 77dc6e296233bed82e5ae809522b90270d05279a Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 8 Oct 2019 16:57:58 +0200
Subject: [PATCH 0494/1013] evl/monitor: call __enter_monitor() unlocked

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/monitor.c | 15 +++++++--------
 1 file changed, 7 insertions(+), 8 deletions(-)

diff --git a/kernel/evl/monitor.c b/kernel/evl/monitor.c
index ad2382638073..fd1c2e126e77 100644
--- a/kernel/evl/monitor.c
+++ b/kernel/evl/monitor.c
@@ -180,7 +180,6 @@ static void wakeup_waiters(struct evl_monitor *event)
 			EVL_MONITOR_TARGETED);
 }
 
-/* nklock held, irqs off */
 static int __enter_monitor(struct evl_monitor *gate,
 			struct evl_monitor_lockreq *req)
 {
@@ -188,6 +187,8 @@ static int __enter_monitor(struct evl_monitor *gate,
 	enum evl_tmode tmode;
 	int info;
 
+	no_ugly_lock();
+
 	if (req) {
 		if ((unsigned long)req->timeout.tv_nsec >= ONE_BILLION)
 			return -EINVAL;
@@ -212,8 +213,6 @@ static int enter_monitor(struct evl_monitor *gate,
 			struct evl_monitor_lockreq *req)
 {
 	struct evl_thread *curr = evl_current();
-	unsigned long flags;
-	int ret;
 
 	no_ugly_lock();
 
@@ -224,11 +223,8 @@ static int enter_monitor(struct evl_monitor *gate,
 		return -EDEADLK; /* Deny recursive locking. */
 
 	evl_commit_monitor_ceiling();
-	xnlock_get_irqsave(&nklock, flags);
-	ret = __enter_monitor(gate, req);
-	xnlock_put_irqrestore(&nklock, flags);
 
-	return ret;
+	return __enter_monitor(gate, req);
 }
 
 static int tryenter_monitor(struct evl_monitor *gate)
@@ -520,8 +516,11 @@ static int wait_monitor(struct file *filp,
 		op_ret = ret;
 	}
 
-	if (ret != -EIDRM)	/* Success or -ETIMEDOUT */
+	if (ret != -EIDRM) {	/* Success or -ETIMEDOUT */
+		xnlock_put_irqrestore(&nklock, flags);
 		ret = __enter_monitor(gate, NULL);
+		goto put;
+	}
 unlock:
 	xnlock_put_irqrestore(&nklock, flags);
 put:
-- 
2.30.2

