From 3ca01abc1a557e981eda0b5a033c0e9ad0912f1b Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 31 Dec 2019 12:22:33 +0100
Subject: [PATCH 547/985] evl/xbuf: fix POLLIN detection

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/xbuf.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/kernel/evl/xbuf.c b/kernel/evl/xbuf.c
index 7f01e74fb..bf8ee9c73 100644
--- a/kernel/evl/xbuf.c
+++ b/kernel/evl/xbuf.c
@@ -210,6 +210,7 @@ static ssize_t do_xbuf_write(struct xbuf_ring *ring,
 	ssize_t len, ret, wbytes, n;
 	unsigned int wroff, avail;
 	unsigned long flags;
+	bool sigpoll;
 	int xret;
 
 	len = wd->count;
@@ -277,9 +278,10 @@ static ssize_t do_xbuf_write(struct xbuf_ring *ring,
 		} while (wbytes > 0);
 
 		if (--ring->wrpending == 0) {
+			sigpoll = ring->fillsz == 0;
 			ring->fillsz += ring->wrrsvd;
 			ring->wrrsvd = 0;
-			ring->signal_input(ring, ring->fillsz == len);
+			ring->signal_input(ring, sigpoll);
 		}
 
 		ring->unlock(ring, flags);
-- 
2.30.2

