From ab58bf3f8ce953a359e88fff3699b68a3b868271 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 22 Jan 2020 12:16:24 +0100
Subject: [PATCH 0559/1013] drivers/evl: add Kconfig help strings

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 drivers/evl/Kconfig | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/drivers/evl/Kconfig b/drivers/evl/Kconfig
index b80e4635695a..792404d29924 100644
--- a/drivers/evl/Kconfig
+++ b/drivers/evl/Kconfig
@@ -1,13 +1,24 @@
 menu "Out-of-band device drivers"
 
 config EVL_LATMUS
-	bool "Latency calibration and measurement"
+	bool "Timer latency calibration and measurement"
 	depends on EVL
 	default y
+	help
+	  This driver supports the latmus application for
+	  determining the best gravity values for the EVL core
+	  clock, and measuring the response time to timer events.
+	  If in doubt, say 'Y'.
 
 config EVL_HECTIC
 	bool "OOB context switching validator"
 	depends on EVL
 	default y
+	help
+	  This driver provides kernel support mainly to the
+	  hectic application, which validates the out-of-band
+	  context switching machinery. It is also required
+	  for running some tests from the EVL test suite.
+	  If in doubt, say 'Y'.
 
 endmenu
-- 
2.30.2

