From 1ac938e9f4b096c3f855fd17280218b297d7f3ff Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 31 Dec 2019 12:26:36 +0100
Subject: [PATCH 0586/1010] evl: drop pointless include directive

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/Makefile | 2 --
 1 file changed, 2 deletions(-)

diff --git a/kernel/evl/Makefile b/kernel/evl/Makefile
index 05cf3e58cf..27d2f4f11d 100644
--- a/kernel/evl/Makefile
+++ b/kernel/evl/Makefile
@@ -1,7 +1,5 @@
 obj-$(CONFIG_EVL) += evl.o sched/
 
-ccflags-y += -Ikernel
-
 evl-y :=		\
 	clock.o		\
 	control.o	\
-- 
2.30.2

