From 32ab4ad6c594e4c963225a7b288d468225e6cfbb Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 31 Mar 2020 17:50:23 +0200
Subject: [PATCH 594/985] evl/thread: switch back out-of-band at trap exit

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/thread.c | 22 +++++++++++++++++++++-
 1 file changed, 21 insertions(+), 1 deletion(-)

diff --git a/kernel/evl/thread.c b/kernel/evl/thread.c
index f5d81701b..1c8f6d236 100644
--- a/kernel/evl/thread.c
+++ b/kernel/evl/thread.c
@@ -1502,7 +1502,7 @@ static inline void note_trap(struct evl_thread *curr,
 }
 
 /* oob stalled. */
-void handle_oob_trap(unsigned int trapnr, struct pt_regs *regs)
+void handle_oob_trap_entry(unsigned int trapnr, struct pt_regs *regs)
 {
 	struct evl_thread *curr;
 	bool is_bp = false;
@@ -1530,8 +1530,28 @@ void handle_oob_trap(unsigned int trapnr, struct pt_regs *regs)
 	 * before handling the exception.
 	 */
 	evl_switch_inband(is_bp ? SIGDEBUG_TRAP : SIGDEBUG_MIGRATE_FAULT);
+}
+
+/* hard irqs on. */
+void handle_oob_trap_exit(unsigned int trapnr, struct pt_regs *regs)
+{
+	struct evl_thread *curr = evl_current();
 
 	curr->local_info &= ~T_INFAULT;
+
+	/*
+	 * Switch back to the oob stage only after recovering from a
+	 * trap in kernel space, which ensures a consistent execution
+	 * state, e.g. if the current task is holding an EVL mutex or
+	 * stax. If the trap occurred in user space, we can leave it
+	 * to the common lazy stage switching strategy since the
+	 * syscall barrier is there to reinstate the proper stage if
+	 * need be.
+	 */
+	if (!user_mode(regs) && !evl_is_breakpoint(trapnr)) {
+		evl_switch_oob();
+		note_trap(curr, trapnr, regs, "resuming out-of-band");
+	}
 }
 
 void handle_oob_mayday(struct pt_regs *regs)
-- 
2.30.2

