From 7219bc9937dfcb3e155f3bcc7734f1cc93c8cd45 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 16 Apr 2020 12:53:16 +0200
Subject: [PATCH 0603/1013] evl/sched: tp: reschedule after control updates

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/sched/tp.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/kernel/evl/sched/tp.c b/kernel/evl/sched/tp.c
index 1690b40237e8..83c5625413b5 100644
--- a/kernel/evl/sched/tp.c
+++ b/kernel/evl/sched/tp.c
@@ -339,11 +339,11 @@ static int tp_control(int cpu, union evl_sched_ctlparam *ctlp,
 	case evl_tp_start:
 		start_tp_schedule(rq);
 		evl_spin_unlock_irqrestore(&rq->lock, flags);
-		return 0;
+		goto done;
 	case evl_tp_stop:
 		stop_tp_schedule(rq);
 		evl_spin_unlock_irqrestore(&rq->lock, flags);
-		return 0;
+		goto done;
 	case evl_tp_get:
 		break;
 	default:
@@ -423,6 +423,8 @@ static int tp_control(int cpu, union evl_sched_ctlparam *ctlp,
 
 	if (ogps)
 		put_tp_schedule(ogps);
+done:
+	evl_schedule();
 
 	return 0;
 fail:
-- 
2.30.2

