From 0c6e45e402127f77b52c642b1916d3a9a3cb25c2 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 29 Jan 2020 18:35:52 +0100
Subject: [PATCH 0604/1010] evl/thread: drop useless thread_info initializer

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/asm-generic/evl/thread_info.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/include/asm-generic/evl/thread_info.h b/include/asm-generic/evl/thread_info.h
index 99cfd71e49..ed2bd105f5 100644
--- a/include/asm-generic/evl/thread_info.h
+++ b/include/asm-generic/evl/thread_info.h
@@ -22,11 +22,6 @@ void evl_init_thread_state(struct oob_thread_state *p)
 
 struct oob_thread_state { };
 
-static inline
-void evl_init_thread_state(struct oob_thread_state *p)
-{
-}
-
 #endif	/* !CONFIG_EVL */
 
 #endif /* !_ASM_GENERIC_EVL_THREAD_INFO_H */
-- 
2.30.2

