From 12ac11030ac360b765e2510d8d27427dadbe20ec Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Mon, 3 Feb 2020 10:05:31 +0100
Subject: [PATCH 0606/1010] evl/flag: add clear operation

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/flag.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/evl/flag.h b/include/evl/flag.h
index 752404bdf1..afebf811a0 100644
--- a/include/evl/flag.h
+++ b/include/evl/flag.h
@@ -128,4 +128,9 @@ static inline void evl_pulse_flag(struct evl_flag *wf)
 	evl_flush_flag(wf, T_BCAST);
 }
 
+static inline void evl_clear_flag(struct evl_flag *wf)
+{
+	wf->raised = false;
+}
+
 #endif /* _EVL_FLAG_H */
-- 
2.30.2

