From 1cf0c87b762319bdd10477d2d8cbbbb7812b45f7 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 10 Jun 2020 11:41:15 +0200
Subject: [PATCH 0629/1013] evl/thread: fix evl_run_kthread() wrapper

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/thread.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/evl/thread.h b/include/evl/thread.h
index 05f41a474ebb..1d300a7e64f9 100644
--- a/include/evl/thread.h
+++ b/include/evl/thread.h
@@ -19,6 +19,7 @@
 #include <evl/list.h>
 #include <evl/lock.h>
 #include <evl/stat.h>
+#include <evl/init.h>
 #include <evl/timer.h>
 #include <evl/sched/param.h>
 #include <evl/factory.h>
@@ -354,7 +355,7 @@ int __evl_run_kthread(struct evl_kthread *kthread, int clone_flags);
 
 #define evl_run_kthread(__kthread, __fn, __priority,			\
 			__clone_flags, __fmt, __args...)		\
-	_evl_run_kthread(__kthread, CPU_MASK_ALL, __fn, __priority,	\
+	_evl_run_kthread(__kthread, &evl_oob_cpus, __fn, __priority,	\
 			__clone_flags, __fmt, ##__args)
 
 #define evl_run_kthread_on_cpu(__kthread, __cpu, __fn, __priority,	\
-- 
2.30.2

