From 63ea5fb552ba7624c73554ecefad3a7833f9bb45 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 8 Apr 2020 19:44:52 +0200
Subject: [PATCH 0631/1010] evl/device: include semaphore header

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/device.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/evl/device.h b/include/evl/device.h
index 3bb85b0c81..afdddc27f4 100644
--- a/include/evl/device.h
+++ b/include/evl/device.h
@@ -16,6 +16,7 @@
 #include <evl/poll.h>
 #include <evl/stax.h>
 #include <evl/work.h>
+#include <evl/sem.h>
 
 #else  /* !CONFIG_EVL */
 
-- 
2.30.2

