From e11da24319da32ddaa6764e131cfc63bdb517381 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Tue, 11 Aug 2020 18:24:35 +0200
Subject: [PATCH 0639/1013] evl/sched: improve naming to clarify traces

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/sched/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/evl/sched/core.c b/kernel/evl/sched/core.c
index 25b214cc88c5..f56585d3a9a6 100644
--- a/kernel/evl/sched/core.c
+++ b/kernel/evl/sched/core.c
@@ -706,7 +706,7 @@ static inline void leave_inband(struct evl_thread *root)
 }
 
 /* oob stalled. */
-static irqreturn_t reschedule_interrupt(int irq, void *dev_id)
+static irqreturn_t oob_reschedule_interrupt(int irq, void *dev_id)
 {
 	trace_evl_reschedule_ipi(this_evl_rq());
 
@@ -1309,7 +1309,7 @@ int __init evl_init_sched(void)
 
 	if (IS_ENABLED(CONFIG_SMP)) {
 		ret = __request_percpu_irq(RESCHEDULE_OOB_IPI,
-					reschedule_interrupt,
+					oob_reschedule_interrupt,
 					IRQF_OOB,
 					"EVL reschedule",
 					&evl_machine_cpudata);
-- 
2.30.2

