From adee60276ad3f8c3fb2c347a7d94012e6a531274 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 10 May 2020 09:54:13 +0200
Subject: [PATCH 0654/1010] evl/work: add evl_cancel_work() helper

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/work.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/include/evl/work.h b/include/evl/work.h
index e3cc4a2a94..84c6ed62c4 100644
--- a/include/evl/work.h
+++ b/include/evl/work.h
@@ -52,6 +52,13 @@ void evl_flush_work(struct evl_work *work)
 	flush_work(&work->wq_work);
 }
 
+static inline
+void evl_cancel_work(struct evl_work *work)
+{
+	irq_work_sync(&work->irq_work);
+	cancel_work_sync(&work->wq_work);
+}
+
 int evl_call_inband_sync_from(struct evl_sync_work *sync_work,
 			struct workqueue_struct *wq);
 
-- 
2.30.2

