From 6de3550ed1ea52680f49fe47fd2cf2eea29cc8c0 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 13 Dec 2020 12:56:12 +0100
Subject: [PATCH 660/985] evl/sched: doc: clarify comment

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/sched/core.c | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/kernel/evl/sched/core.c b/kernel/evl/sched/core.c
index cdf77302d..ea2547891 100644
--- a/kernel/evl/sched/core.c
+++ b/kernel/evl/sched/core.c
@@ -887,10 +887,11 @@ static inline bool test_resched(struct evl_rq *this_rq)
 }
 
 /*
- * CAUTION: curr->altsched.task may be unsynced and even stale if curr
- * == &this_rq->root_thread, since the task logged by leave_inband()
- * may not still be the current one. Use "current" for disambiguating
- * if you need to refer to the underlying inband task.
+ * CAUTION: curr->altsched.task may be unsynced and even stale if
+ * (this_rq->curr == &this_rq->root_thread), since the task logged by
+ * dovetail_context_switch() may not still be the current one. Always
+ * use "current" for disambiguating if you intend to refer to the
+ * running inband task.
  */
 void __evl_schedule(void) /* oob or oob stalled (CPU migration-safe) */
 {
-- 
2.30.2

