From b66a250f1463f51971a7e44f71715752a154b333 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 3 Sep 2020 17:08:35 +0200
Subject: [PATCH 0680/1010] evl/assert: add explicit dependency on irqstage
 definitions

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/assert.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/evl/assert.h b/include/evl/assert.h
index 842c8e430a..e4cd8a47c9 100644
--- a/include/evl/assert.h
+++ b/include/evl/assert.h
@@ -9,6 +9,7 @@
 #define _EVL_ASSERT_H
 
 #include <linux/kconfig.h>
+#include <linux/irqstage.h>
 
 #define EVL_INFO	KERN_INFO    "EVL: "
 #define EVL_WARNING	KERN_WARNING "EVL: "
-- 
2.30.2

