From fc7d2a6bdaf18c06a3229a77d4c9426b371f9f57 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 11 Sep 2021 17:02:22 +0200
Subject: [PATCH 0746/1010] evl/monitor: remove unused variable

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/monitor.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/kernel/evl/monitor.c b/kernel/evl/monitor.c
index d19bd0e7d2..e0b31ecda4 100644
--- a/kernel/evl/monitor.c
+++ b/kernel/evl/monitor.c
@@ -38,7 +38,6 @@ struct evl_monitor {
 			struct evl_monitor *gate;
 			struct evl_poll_head poll_head;
 			struct list_head next; /* in ->events */
-			struct list_head next_poll;
 		};
 	};
 };
-- 
2.30.2

