From 2bbd762c7c7d16e21391ea4145292677a3c766cf Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 13 Nov 2021 16:04:57 +0100
Subject: [PATCH 0755/1010] evl: kconfig: turn on DOVETAIL_DEBUG on CORE/NET
 debug

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/Kconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/evl/Kconfig b/kernel/evl/Kconfig
index 811deba7dc..be00a495fa 100644
--- a/kernel/evl/Kconfig
+++ b/kernel/evl/Kconfig
@@ -235,6 +235,7 @@ if EVL_DEBUG
 
 config EVL_DEBUG_CORE
 	bool "Core runtime assertions"
+	select DOVETAIL_DEBUG
 	help
 	  This option activates various assertions inside the EVL
 	  core. This option has moderate overhead.
@@ -258,6 +259,7 @@ config EVL_DEBUG_WOLI
 
 config EVL_DEBUG_NET
 	bool "Network stack debugging"
+	select DOVETAIL_DEBUG
 	help
 	  This option activates various assertions inside the EVL
 	  network stack. This option has moderate overhead.
-- 
2.30.2

