From a314bfa4bc8e94cdd4e5d0d38b64a7dd6a95c4e1 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 3 Dec 2022 12:52:18 +0100
Subject: [PATCH 0802/1013] evl/wait: drop debug leftover

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/evl/wait.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/evl/wait.h b/include/evl/wait.h
index 74312448c839..db54e5587016 100644
--- a/include/evl/wait.h
+++ b/include/evl/wait.h
@@ -32,10 +32,10 @@ struct evl_wait_channel {
 struct evl_lock_key_addr {
 	struct lock_class_key *addr;
 };
-#define __EVL_LOCK_KEY_ADDR_INITIALIZER  (struct foo){ .addr = NULL }
+#define __EVL_LOCK_KEY_ADDR_INITIALIZER  (struct evl_lock_key_addr){ .addr = NULL }
 #else
 struct evl_lock_key_addr { };
-#define __EVL_LOCK_KEY_ADDR_INITIALIZER  (struct foo){ }
+#define __EVL_LOCK_KEY_ADDR_INITIALIZER  (struct evl_lock_key_addr){ }
 #endif
 
 struct evl_wait_queue {
-- 
2.30.2

