From b52d0cb9305ab03229c7a37ed1d54199bf95e877 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Thu, 24 Aug 2023 12:28:19 +0200
Subject: [PATCH 0821/1013] spidev: evl: include missing header

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 include/uapi/evl/devices/spidev.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/uapi/evl/devices/spidev.h b/include/uapi/evl/devices/spidev.h
index a4b3f97f384c..78ab4e709684 100644
--- a/include/uapi/evl/devices/spidev.h
+++ b/include/uapi/evl/devices/spidev.h
@@ -5,6 +5,8 @@
 #ifndef _EVL_UAPI_DEVICES_SPIDEV_H
 #define _EVL_UAPI_DEVICES_SPIDEV_H
 
+#include <linux/types.h>
+
 /* Manage out-of-band mode (master only) */
 struct spi_ioc_oob_setup {
 	/* Input */
-- 
2.30.2

