From afbe4b3e6a0320c4f5aa9761be473b797aca0ca9 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sun, 4 Dec 2022 12:41:53 +0100
Subject: [PATCH 0840/1010] evl/monitor: display monitor name in state
 attribute

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/monitor.c | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/kernel/evl/monitor.c b/kernel/evl/monitor.c
index c1abc20296..fd9841f69e 100644
--- a/kernel/evl/monitor.c
+++ b/kernel/evl/monitor.c
@@ -968,17 +968,23 @@ static ssize_t state_show(struct device *dev,
 		}
 	} else {
 		fun = atomic_read(&state->u.gate.owner);
-		if (fun != EVL_NO_HANDLE)
+		if (fun != EVL_NO_HANDLE) {
 			owner = evl_get_factory_element_by_fundle(&evl_thread_factory,
 						evl_get_index(fun),
 						struct evl_thread);
-		ret = snprintf(buf, PAGE_SIZE, "%d %u %u\n",
-			owner ? evl_get_inband_pid(owner) : -1,
-			state->u.gate.ceiling,
-			owner ? (state->u.gate.recursive ?
-				state->u.gate.nesting : 1) : 0);
-		if (owner)
+			if (!owner)
+				goto no_owner;
+			ret = snprintf(buf, PAGE_SIZE, "%s(%d) %u %u\n",
+				evl_element_name(&owner->element),
+				evl_get_inband_pid(owner),
+				state->u.gate.ceiling,
+				state->u.gate.recursive ? state->u.gate.nesting : 1);
 			evl_put_element(&owner->element);
+		} else {
+		no_owner:
+			ret = snprintf(buf, PAGE_SIZE, "-1 %u 0\n",
+				state->u.gate.ceiling);
+		}
 	}
 
 	evl_put_element(&mon->element);
-- 
2.30.2

