From c7a9c7a2d6c4a65a7f6e1ff85ce96445756c3d38 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Tue, 13 Feb 2024 15:23:06 +0100
Subject: [PATCH 872/985] evl/net: socket: fixup signature of bind, shutdown
 handlers

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 kernel/evl/net/socket.c | 44 ++++++++++++++++++++---------------------
 1 file changed, 22 insertions(+), 22 deletions(-)

diff --git a/kernel/evl/net/socket.c b/kernel/evl/net/socket.c
index 28cbe91dc..8b7bda8fc 100644
--- a/kernel/evl/net/socket.c
+++ b/kernel/evl/net/socket.c
@@ -339,63 +339,63 @@ void sock_oob_destroy(struct sock *sk)
  * binding of the network socket to the given address by the in-band
  * stack.
  */
-int sock_oob_bind(struct socket *sock, struct sockaddr *addr, int len)
+int sock_oob_bind(struct sock *sk, struct sockaddr *addr, int len)
 {
-	struct evl_socket *esk = evl_sk(sock->sk);
+	struct evl_socket *esk = evl_sk(sk);
 
 	/*
 	 * If @sock belongs to PF_OOB, then evl_sock_bind() already
-	 * handled the binding. We ony care about common protocols
+	 * handled the binding. We only care about common protocols
 	 * for which we have an out-of-band extension
 	 * (e.g. AF_PACKET).
 	 */
-	if (sock->sk->sk_family == PF_OOB || !esk->proto->bind)
+	if (sk->sk_family == PF_OOB || !esk->proto->bind)
 		return 0;
 
 	return esk->proto->bind(esk, addr, len);
 }
 
 /*
- * In-band call from the common network stack to connect the socket.
- * We end up here _after_ a successful connection of the network
- * socket to the given address by the in-band stack.
+ * In-band call from the common network stack to shutdown the
+ * socket. We end up here _after_ the socket was successfully shut
+ * down by the in-band network stack.
  */
-int sock_oob_connect(struct socket *sock,
-		struct sockaddr *addr, int len, int flags)
+int sock_oob_shutdown(struct sock *sk, int how)
 {
-	struct evl_socket *esk = evl_sk(sock->sk);
+	struct evl_socket *esk = evl_sk(sk);
 
 	/*
-	 * If @sock belongs to PF_OOB, then evl_sock_connect() already
-	 * handled the connection. We ony care about common protocols
+	 * If @sock belongs to PF_OOB, then evl_sock_shutdown() already
+	 * handled the connection. We only care about common protocols
 	 * for which we have an out-of-band extension
 	 * (e.g. AF_INET/IPPROTO_UDP).
 	 */
-	if (sock->sk->sk_family == PF_OOB || !esk->proto->connect)
+	if (sk->sk_family == PF_OOB || !esk->proto->shutdown)
 		return 0;
 
-	return esk->proto->connect(esk, addr, len, flags);
+	return esk->proto->shutdown(esk, how);
 }
 
 /*
- * In-band call from the common network stack to shutdown the
- * socket. We end up here _after_ the socket was successfully shut
- * down by the in-band network stack.
+ * In-band call from the common network stack to connect the socket.
+ * We end up here _after_ a successful connection of the network
+ * socket to the given address by the in-band stack.
  */
-int sock_oob_shutdown(struct socket *sock, int how)
+int sock_oob_connect(struct socket *sock,
+		struct sockaddr *addr, int len, int flags)
 {
 	struct evl_socket *esk = evl_sk(sock->sk);
 
 	/*
-	 * If @sock belongs to PF_OOB, then evl_sock_shutdown() already
-	 * handled the connection. We ony care about common protocols
+	 * If @sock belongs to PF_OOB, then evl_sock_connect() already
+	 * handled the connection. We only care about common protocols
 	 * for which we have an out-of-band extension
 	 * (e.g. AF_INET/IPPROTO_UDP).
 	 */
-	if (sock->sk->sk_family == PF_OOB || !esk->proto->shutdown)
+	if (sock->sk->sk_family == PF_OOB || !esk->proto->connect)
 		return 0;
 
-	return esk->proto->shutdown(esk, how);
+	return esk->proto->connect(esk, addr, len, flags);
 }
 
 static int socket_send_recv(struct evl_socket *esk,
-- 
2.30.2

