From d0575632b7f40f3f38c51a249ccbf5407b9033f3 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 1 Feb 2025 12:55:59 +0100
Subject: [PATCH 0906/1013] drivers/evl: hectic: release cdev region on unload

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 drivers/evl/hectic.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/evl/hectic.c b/drivers/evl/hectic.c
index 4cff66cb1e0c..0112f636008d 100644
--- a/drivers/evl/hectic.c
+++ b/drivers/evl/hectic.c
@@ -710,6 +710,7 @@ static void __exit hectic_exit(void)
 {
 	device_destroy(&hectic_class, MKDEV(MAJOR(hectic_devt), 0));
 	cdev_del(&hectic_cdev);
+	unregister_chrdev_region(hectic_devt, 1);
 	class_unregister(&hectic_class);
 }
 module_exit(hectic_exit);
-- 
2.30.2

