From f5c4db42bde43bcb9c3c25ad8e10625c7ef1f698 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Sun, 27 Apr 2025 16:08:45 +0200
Subject: [PATCH 919/985] evl/sched: use regular format for CPU id

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 kernel/evl/sched/core.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/evl/sched/core.c b/kernel/evl/sched/core.c
index 7154481b7..c1da116dc 100644
--- a/kernel/evl/sched/core.c
+++ b/kernel/evl/sched/core.c
@@ -107,10 +107,10 @@ static void watchdog_handler(struct evl_timer *timer) /* oob stage stalled */
 		raw_spin_unlock(&curr->lock);
 		evl_notify_thread(curr, EVL_HMDIAG_WATCHDOG, evl_nil);
 		dovetail_send_mayday(current);
-		printk(EVL_WARNING "watchdog triggered on CPU #%d -- runaway thread "
+		printk(EVL_WARNING "watchdog triggered on CPU%d -- runaway thread "
 			"'%s' signaled\n", evl_rq_cpu(this_rq), curr->name);
 	} else {
-		printk(EVL_WARNING "watchdog triggered on CPU #%d -- runaway thread "
+		printk(EVL_WARNING "watchdog triggered on CPU%d -- runaway thread "
 			"'%s' canceled\n", evl_rq_cpu(this_rq), curr->name);
 		/*
 		 * On behalf on an IRQ handler, evl_cancel_thread()
-- 
2.30.2

