From 10a0ecefe12dacc3e46097ac6518ea6d85ac7a00 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Fri, 2 May 2025 11:01:37 +0200
Subject: [PATCH 0924/1013] evl/net: skb: unionize exclusive fields in control
 block

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 include/evl/net/skb.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/include/evl/net/skb.h b/include/evl/net/skb.h
index 9103292f242c..593ac759485b 100644
--- a/include/evl/net/skb.h
+++ b/include/evl/net/skb.h
@@ -20,9 +20,11 @@ struct evl_socket;
 struct iovec;
 
 struct evl_net_cb {
-	struct evl_net_handler *handler;
-	struct evl_socket *tracker;
 	struct net_device *dev;
+	union {
+		struct evl_net_handler *handler;
+		struct evl_socket *tracker;
+	};
 	union {
 		/* protocol-specific stuff should live here. */
 	};
-- 
2.30.2

