From c3fe941470a293de6d6696f0b1d9bd9e8a7b2001 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Thu, 5 Jun 2025 09:26:18 +0200
Subject: [PATCH 937/985] evl/net: ipv4: do not solicit non-ARP neighbours

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 kernel/evl/net/ipv4/ipv4.c | 28 ++++++++++++++++------------
 1 file changed, 16 insertions(+), 12 deletions(-)

diff --git a/kernel/evl/net/ipv4/ipv4.c b/kernel/evl/net/ipv4/ipv4.c
index 42e8bbb16..0289e01ec 100644
--- a/kernel/evl/net/ipv4/ipv4.c
+++ b/kernel/evl/net/ipv4/ipv4.c
@@ -242,7 +242,7 @@ int evl_net_ipv4_solicit(struct evl_socket *esk,
 	struct net_device *dev;
 	struct rtable *rt;
 	__be32 ipaddr;
-	long ret;
+	long ret = 0;
 
 	if (addr->sa_family != AF_INET)
 		return -EAFNOSUPPORT;
@@ -256,10 +256,10 @@ int evl_net_ipv4_solicit(struct evl_socket *esk,
 	if (IS_ERR(rt))
 		return PTR_ERR(rt);
 
-	if (!rt->dst.dev || !netif_oob_port(rt->dst.dev))
+	dev = rt->dst.dev;
+	if (!dev || !netif_oob_port(dev))
 		return -ENODEV;
 
-	dev = rt->dst.dev;
 	evl_net_get_dev(dev);
 
 	neigh = dst_neigh_lookup(&rt->dst, &ipaddr);
@@ -269,29 +269,33 @@ int evl_net_ipv4_solicit(struct evl_socket *esk,
 		return -ENOENT;
 	}
 
-	if (!(neigh->nud_state & (NUD_VALID & ~NUD_STALE)))
-		neigh_event_send(neigh, NULL);
+	if (likely(!(neigh->nud_state & NUD_NOARP))) {
+		if (!(neigh->nud_state & (NUD_VALID & ~NUD_STALE)))
+			neigh_event_send(neigh, NULL);
+
+		/* Wait for the ARP entry to enter the cache. */
+		ret = wait_event_interruptible_timeout(evl_arp_event,
+						(e = evl_net_get_arp_entry(dev, ipaddr)),
+						HZ * 5);
+	}
 
-	/* Wait for the ARP entry to enter the cache. */
-	ret = wait_event_interruptible_timeout(evl_arp_event,
-				(e = evl_net_get_arp_entry(dev, ipaddr)),
-				HZ * 5);
 	evl_net_put_dev(dev);
 
 	if (e) {
 		/* We never downgrade the permanent state. */
 		if (flags & EVL_NEIGH_PERMANENT &&
-			!(neigh->nud_state & NUD_PERMANENT)) {
+			!(neigh->nud_state & NUD_PERMANENT))
 			ret = neigh_update(neigh, e->ha, NUD_PERMANENT,
 				NEIGH_UPDATE_F_OVERRIDE | NEIGH_UPDATE_F_ADMIN, 0);
-		}
 		evl_net_put_arp_entry(e);
 		ret = 0;
+	} else {
+		ret = -ETIMEDOUT;
 	}
 
 	neigh_release(neigh);
 
-	return ret <= 0 ? ret : -ETIMEDOUT;
+	return ret;
 }
 
 static struct evl_net_proto *match_ipv4_domain(int type, int protocol)
-- 
2.30.2

