From 25e14c50a665bc58411bfd1345a1d6f047f89b57 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 11 Jun 2025 16:49:03 +0200
Subject: [PATCH 0946/1013] evl/net: ipv4: solicit via global routing

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 kernel/evl/net/ipv4/ipv4.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/evl/net/ipv4/ipv4.c b/kernel/evl/net/ipv4/ipv4.c
index 581353b4e48c..b44138922ba5 100644
--- a/kernel/evl/net/ipv4/ipv4.c
+++ b/kernel/evl/net/ipv4/ipv4.c
@@ -252,7 +252,7 @@ int evl_net_ipv4_solicit(struct evl_socket *esk,
 	if (flags & ~EVL_NEIGH_PERMANENT)
 		return -EINVAL;
 
-	rt = ip_route_output(sock_net(esk->sk), ipaddr, 0, 0, 0, RT_SCOPE_LINK);
+	rt = ip_route_output(sock_net(esk->sk), ipaddr, 0, 0, 0, RT_SCOPE_UNIVERSE);
 	if (IS_ERR(rt))
 		return PTR_ERR(rt);
 
-- 
2.30.2

