From 8560953cbc211b15ebde44e852c80c1d00b61891 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Fri, 2 May 2025 16:03:09 +0200
Subject: [PATCH 0951/1010] evl/net: skb: allow for data-less buffers on
 release

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 kernel/evl/net/skb.c | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/kernel/evl/net/skb.c b/kernel/evl/net/skb.c
index 5c8b0b642e..ea7f3d3d2c 100644
--- a/kernel/evl/net/skb.c
+++ b/kernel/evl/net/skb.c
@@ -218,15 +218,16 @@ static void __free_evl_skb(struct sk_buff *skb, struct net_device *dev)
 	if (skb->cloned &&
 	    atomic_sub_return(skb->nohdr ? (1 << SKB_DATAREF_SHIFT) + 1 : 1,
 			      &shinfo->dataref))
-		goto release_head;
+		goto put_skb;
 
 	/*
-	 * Release the data. This is the gist of skb_pp_recycle(),
-	 * since we already know for sure that an oob-managed skb is
-	 * built around a page from a per-device pool (in
-	 * evl_netdev_state).
+	 * Attempt to release the heading data to the originating page
+	 * pool if any.
 	 */
-	page_pool_return_skb_page(virt_to_page(skb->head));
+	if (unlikely(!skb->head))
+		goto put_skb;
+
+	skb_pp_recycle(skb, skb->head);
 
 	/*
 	 * Wake up any thread waiting for buffer space to send to the
@@ -241,9 +242,8 @@ static void __free_evl_skb(struct sk_buff *skb, struct net_device *dev)
 
 	evl_signal_poll_events(&est->poll_head,	POLLOUT|POLLWRNORM);
 
-release_head:
+put_skb:
 	EVL_WARN_ON(NET, atomic_read(&shinfo->dataref) < 0);
-	/* Now release the buffer head. */
 	put_oob_skb(skb);
 }
 
@@ -344,14 +344,14 @@ void evl_net_free_skb_list(struct list_head *list)
 }
 
 /**
- *	free_skb_oob - attempt to free a buffer head along with its
- *	data storage.
+ *	free_skb_oob - attempt to free a buffer along with its data
+ *	storage.
  *
  *      Called from the in-band net core right after the last
  *      reference to the buffer was dropped. We get a chance to
- *      release the buffer immediately to our bufheads pool. However,
- *      if the buffer data was allocated in-band, send the skb back to
- *      the in-band core for disposal from there.
+ *      release the buffer immediately to our pool. However, if the
+ *      buffer data was allocated in-band, send the skb back to the
+ *      in-band core for disposal from there.
  */
 void free_skb_oob(struct sk_buff *skb) /* inband/oob */
 {
-- 
2.30.2

