From 728b86c0a1964f1a5f6b81b6e3f9807b79ec640e Mon Sep 17 00:00:00 2001
From: Giulio Moro <giuliomoro@yahoo.it>
Date: Wed, 8 Oct 2025 13:28:29 +0000
Subject: [PATCH 0986/1013] Remove stale CONTEXT_USER (now CT_STATE_USER)

Signed-off-by: Giulio Moro <giuliomoro@yahoo.it>
---
 arch/arm64/kernel/entry-common.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/arm64/kernel/entry-common.c b/arch/arm64/kernel/entry-common.c
index 52a995f9016b..125de5b0984e 100644
--- a/arch/arm64/kernel/entry-common.c
+++ b/arch/arm64/kernel/entry-common.c
@@ -171,7 +171,6 @@ static __always_inline void __enter_from_user_mode(void)
 		lockdep_hardirqs_off(CALLER_ADDR0);
 		WARN_ON_ONCE(irq_pipeline_debug() && test_inband_stall());
 		CT_WARN_ON(ct_state() != CT_STATE_USER);
-		CT_WARN_ON(ct_state() != CONTEXT_USER);
 		stall_inband_nocheck();
 		user_exit_irqoff();
 		unstall_inband_nocheck();
-- 
2.30.2

