From f777cd299e3bcd0cf202c9d0309aa35ce861c419 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <philippe.gerum@exail.com>
Date: Fri, 27 Jun 2025 16:09:58 +0200
Subject: [PATCH 1004/1010] evl: bump to ABI 42

Signed-off-by: Philippe Gerum <philippe.gerum@exail.com>
---
 include/uapi/evl/control-abi.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/uapi/evl/control-abi.h b/include/uapi/evl/control-abi.h
index f2137a4772..1ba6132733 100644
--- a/include/uapi/evl/control-abi.h
+++ b/include/uapi/evl/control-abi.h
@@ -17,7 +17,7 @@
  * latter to the former. CAUTION: a litteral value is required for the
  * current ABI definition (scripts reading this may be naive).
  */
-#define EVL_ABI_LEVEL  41
+#define EVL_ABI_LEVEL  42
 
 #define EVL_CONTROL_DEV  "/dev/evl/control"
 
-- 
2.30.2

