From 723a7e5d5e29c63ae587827c7f6b97b08d69a2f7 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Wed, 17 Sep 2025 10:02:45 +0200
Subject: [PATCH 1006/1010] evl/syscall: add missing header for EL0 compat mode

Signed-off-by: Philippe Gerum <rpm@xenomai.org>
---
 arch/arm64/include/asm/evl/syscall.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/include/asm/evl/syscall.h b/arch/arm64/include/asm/evl/syscall.h
index 7459d82c35..175f37e335 100644
--- a/arch/arm64/include/asm/evl/syscall.h
+++ b/arch/arm64/include/asm/evl/syscall.h
@@ -3,6 +3,7 @@
 #define _EVL_ARM64_ASM_SYSCALL_H
 
 #include <linux/uaccess.h>
+#include <asm/compat.h>
 #include <asm/unistd.h>
 #include <asm/ptrace.h>
 #include <uapi/asm-generic/dovetail.h>
-- 
2.30.2

